core/domain/learner_group_services.py

```diff
@@ -735,6 +737,33 @@ def remove_story_reference_from_learner_groups(story_id: str) -> None:
     learner_group_models.LearnerGroupModel.put_multi(models_to_put)
 
 
+def get_model_remove_story_reference(
+    story_id: str
+) -> List[base_models.BaseModel]:
+    """Removes a given story id from all learner groups that have it's
+    reference.
+
+    Args:
+        story_id: str. Story id to remove.
+
+    Returns:
+        list(BaseModel). List of BaseModel.
+    """
+    found_models: Sequence[learner_group_models.LearnerGroupModel] = (
```

Comment:
Generally speaking, this isn't the right approach, because we don't pass models around in the domain layer. Can you have the caller of this function fetch the domain objects corresponding to the models, then do a for loop on those domain objects to drop the story_id, then create the necessary models from the domain objects just before they are saved to the datastore?